auth/basics/android.md

```diff
@@ -46,19 +53,45 @@ skygear.signupWithEmail(email, password, new AuthResponseHandler() {
     }
 
     @Override
-    public void onAuthFail(String reason) {
-        Log.i("Skygear Signup", "onAuthFail: Fail with reason: " + reason);
+    public void onAuthFail(Error error) {
+        Log.i("Skygear Signup", "onAuthFail: Fail with reason: " + error.getMessage());
     }
 });
 ```
 
-## Logging in
+### Anonymous user
 
-The following shows how to log in a user with username and password.
+Skygear support users to sign up anonymously. However, they can only read data from the public database, but cannot perform operations such as saving data.
+
+To create an anonymous user, call `skygear.signupAnonymously()` as shown below:
 
 ```java
-Container skygear = Container.defaultContainer(this);
+skygear.signupAninymously(new AuthResponseHandler() {
+    @Override
+    public void onAuthSuccess(User user) {
+        Log.i("Skygear Signup", "onAuthSuccess: Got token: " + user.accessToken);
+    }
+
+    @Override
+    public void onAuthFail(Error error) {
+        Log.i("Skygear Signup", "onAuthFail: Fail with reason: " + error.getMessage());
+    }
+});
+```
+
+Each anonymous user has a unique ID that behaves exactly the same way as normal. As expected, an anonymous user does not have username, email nor password.
+
+Because of this behaviour, once the token of an anonymous user is lost, so will the account.
```

Comment:
To me, the word "lost" is a little ambiguous here. The user, the account, and the saved records will persist, just that the user cannot log in as that anonymous user again